File: OPENAPI-REST-API/openapi-client/scala-gatling/src/gatling/scala/org/openapitools/client/model/Website.scala

package org.openapitools.client.model


case class Website (
    _serviceInfo: Option[WebsiteServiceInfo],
    _client_links: Option[List[WebsiteClientLink]],
    _billingDetails: Option[WebsiteBillingDetails],
    /* Customer's currency */
    _custCurrency: Option[String],
    /* Customer currency symbol */
    _custCurrencySymbol: Option[String],
    _serviceMaster: Option[WebsiteServiceMaster],
    /* Package information */
    _package: Option[String],
    _serviceExtra: Option[List[Any]],
    _extraInfoTables: Option[WebsiteExtraInfoTables]
)
object Website {
    def toStringBody(var_serviceInfo: Object, var_client_links: Object, var_billingDetails: Object, var_custCurrency: Object, var_custCurrencySymbol: Object, var_serviceMaster: Object, var_package: Object, var_serviceExtra: Object, var_extraInfoTables: Object) =
        s"""
        | {
        | "serviceInfo":$var_serviceInfo,"client_links":$var_client_links,"billingDetails":$var_billingDetails,"custCurrency":$var_custCurrency,"custCurrencySymbol":$var_custCurrencySymbol,"serviceMaster":$var_serviceMaster,"package":$var_package,"serviceExtra":$var_serviceExtra,"extraInfoTables":$var_extraInfoTables
        | }
        """.stripMargin
}

File: OPENAPI-REST-API/openapi-client/zapier/models/Website.js
const utils = require('../utils/utils');
const WebsiteBillingDetails = require('../models/WebsiteBillingDetails');
const WebsiteClientLink = require('../models/WebsiteClientLink');
const WebsiteExtraInfoTables = require('../models/WebsiteExtraInfoTables');
const WebsiteServiceInfo = require('../models/WebsiteServiceInfo');
const WebsiteServiceMaster = require('../models/WebsiteServiceMaster');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            ...WebsiteServiceInfo.fields(`${keyPrefix}serviceInfo`, isInput),
            {
                key: `${keyPrefix}client_links`,
                label: `[${labelPrefix}client_links]`,
                children: WebsiteClientLink.fields(`${keyPrefix}client_links${!isInput ? '[]' : ''}`, isInput, true), 
            },
            ...WebsiteBillingDetails.fields(`${keyPrefix}billingDetails`, isInput),
            {
                key: `${keyPrefix}custCurrency`,
                label: `Customer's currency - [${labelPrefix}custCurrency]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}custCurrencySymbol`,
                label: `Customer currency symbol - [${labelPrefix}custCurrencySymbol]`,
                type: 'string',
            },
            ...WebsiteServiceMaster.fields(`${keyPrefix}serviceMaster`, isInput),
            {
                key: `${keyPrefix}package`,
                label: `Package information - [${labelPrefix}package]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}serviceExtra`,
                label: `[${labelPrefix}serviceExtra]`,
                list: true,
                type: 'object',
            },
            ...WebsiteExtraInfoTables.fields(`${keyPrefix}extraInfoTables`, isInput),
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'serviceInfo': utils.removeIfEmpty(WebsiteServiceInfo.mapping(bundle, `${keyPrefix}serviceInfo`)),
            'client_links': utils.childMapping(bundle.inputData?.[`${keyPrefix}client_links`], `${keyPrefix}client_links`, WebsiteClientLink),
            'billingDetails': utils.removeIfEmpty(WebsiteBillingDetails.mapping(bundle, `${keyPrefix}billingDetails`)),
            'custCurrency': bundle.inputData?.[`${keyPrefix}custCurrency`],
            'custCurrencySymbol': bundle.inputData?.[`${keyPrefix}custCurrencySymbol`],
            'serviceMaster': utils.removeIfEmpty(WebsiteServiceMaster.mapping(bundle, `${keyPrefix}serviceMaster`)),
            'package': bundle.inputData?.[`${keyPrefix}package`],
            'serviceExtra': bundle.inputData?.[`${keyPrefix}serviceExtra`],
            'extraInfoTables': utils.removeIfEmpty(WebsiteExtraInfoTables.mapping(bundle, `${keyPrefix}extraInfoTables`)),
        }
    },
}
